onnxruntime/core/providers/cpu/sequence/sequence_ops.cc

```diff
@@ -316,4 +319,229 @@ Status SequenceConstruct::Compute(OpKernelContext* context) const {
 
   return Status::OK();
 }
+
+// SplitToSequence
+ONNX_CPU_OPERATOR_KERNEL(
+    SplitToSequence,
+    11,
+    KernelDefBuilder().TypeConstraint("T",
+                                      std::vector<MLDataType>{
+                                          DataTypeImpl::GetTensorType<float>(),
+                                          DataTypeImpl::GetTensorType<int32_t>(),
+                                          DataTypeImpl::GetTensorType<std::string>()})
+        .TypeConstraint("S", DataTypeImpl::AllSequenceTensorTypes())
+        .TypeConstraint("I", std::vector<MLDataType>{
+                                 DataTypeImpl::GetTensorType<int32_t>(),
+                                 DataTypeImpl::GetTensorType<int64_t>()}),
+    SplitToSequence);
+
+SplitToSequence::SplitToSequence(const OpKernelInfo& info) : OpKernel(info) {
+  axis_ = info.GetAttrOrDefault<int64_t>("axis", 0);
+  keepdims_ = info.GetAttrOrDefault<int64_t>("keepdims", 1);
+}
+
+Status SplitToSequence::Compute(OpKernelContext* context) const {
+  const Tensor& input = *context->Input<Tensor>(0);
+  const Tensor* p_split_input = context->Input<Tensor>(1);
+
+  Status status;
+  auto data_type = input.DataType();
+
+  if (data_type == DataTypeImpl::GetType<float>())
+    status = ComputeImpl<float>(*context, input, p_split_input);
+  else if (data_type == DataTypeImpl::GetType<int32_t>())
+    status = ComputeImpl<int32_t>(*context, input, p_split_input);
+  else if (data_type == DataTypeImpl::GetType<std::string>())
+    status = ComputeImpl<std::string>(*context, input, p_split_input);
+  else
+    ORT_THROW("SplitToSequence operator does not support ", data_type, " yet");
+
+  return status;
+}
+
+Status SplitToSequence::PrepareForCompute(const TensorShape& input_shape, int64_t split_scalar, bool is_split_input_scalar,
+                                          int64_t& num_outputs, int64_t& axis, int& before_dims,
+                                          int& after_dims_including_split_axis, int& after_dims_excluding_split,
+                                          bool& is_uneven_split, int& num_remaining_splits,
+                                          std::vector<int64_t>& split_sizes) const {
+  auto& input_dims = input_shape.GetDims();
+  const auto num_dimensions = gsl::narrow_cast<int64_t>(input_shape.NumDimensions());
+  axis = HandleNegativeAxis(axis_, num_dimensions);  // handle negative and enforce axis is valid
+  const int64_t split_dim_size = input_dims[axis];
+
+  before_dims = gsl::narrow<int>(input_shape.SizeToDimension(axis));
+  after_dims_including_split_axis = gsl::narrow<int>(input_shape.SizeFromDimension(axis));
+  after_dims_excluding_split = (axis + 1 == num_dimensions)
+                                   ? 1  // we multiply by this value so must be 1 not 0
+                                   : gsl::narrow<int>(input_shape.SizeFromDimension(axis + 1));
```

Comment:
from https://github.com/microsoft/onnxruntime/blob/bd2d6af9ca47f21880fe121d3a7c5392f77d9826/onnxruntime/core/framework/tensor_shape.cc#L42 and https://github.com/microsoft/onnxruntime/blob/bd2d6af9ca47f21880fe121d3a7c5392f77d9826/onnxruntime/core/framework/tensor_shape.cc#L83 it would seem the result is 1 as well for the else branch

Reply:
This was derived from the Split op code [here](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cpu/tensor/split.cc#L35).